zstd-safe/zstd-sys/build.rs

```diff
@@ -101,22 +101,21 @@ fn compile_zstd() {
         let mut entries: Vec<_> = fs::read_dir(dir)
             .unwrap()
             .map(|r| r.unwrap().path())
-            .collect();
-        entries.sort();
-        for path in entries {
+            .filter(|path| path.extension() == Some(OsStr::new("c")))
             // Skip xxhash*.c files: since we are using the "PRIVATE API"
             // mode, it will be inlined in the headers.
-            if path
-                .file_name()
-                .and_then(|p| p.to_str())
-                .map_or(false, |p| p.contains("xxhash"))
-            {
-                continue;
-            }
-            if path.extension() == Some(OsStr::new("c")) {
-                config.file(path);
-            }
-        }
+            .filter(|path| {
+               !path
+                    .file_name()
+                    // If path.extension() is Some, then file_name cannot be None.
+                    .unwrap()
+                    .to_string_lossy()
```

Comment:
We could also use `as_bytes()` to get the backing `&[u8]` and then look for `b"xxhash"` in there, so we avoid allocation even when it's not valid utf-8.

Reply:
Nevermind i thought there was such a contains function.